v2/AdminCore.Services/DashboardService.cs

```diff
@@ -3,19 +3,32 @@
 using AdminCore.DTOs.Event;
 using System;
 using System.Collections.Generic;
+using System.Linq;
+using AdminCore.DAL;
+using AdminCore.Services.Base;
+using AutoMapper;
 
 namespace AdminCore.Services
 {
-  public class DashboardService : IDashboardService
+  public class DashboardService : BaseService, IDashboardService
   {
+    private readonly IMapper _mapper;
+    private readonly IEventService _eventService;
+
+    public DashboardService(IDatabaseContext databaseContext, IEventService eventService, IMapper mapper) : base(databaseContext)
+    {
+      _mapper = mapper;
+      _eventService = eventService;
+    }
     public IList<EventDto> GetEmployeeDashboardEvents(int employeeId, DateTime date)
     {
       throw new NotImplementedException();
     }
 
-    public IList<EventDto> GetEmployeeEvents(int employeeId, DateTime date)
+    public IList<EventDto> GetEmployeeEventsForAGivenMonth(int employeeId, DateTime date)
     {
-      throw new NotImplementedException();
+      var eventsForEmployee = _eventService.GetEventsByEmployeeId(employeeId);
+      return eventsForEmployee.Where(employeeEvent => EventHappensThisMonth(employeeEvent, date)).ToList();
```

Comment:
So here we’re Retrieving all events for an employee and then delimiting in memory, it should be done at the database level

Reply:
Fixed